packages/flutter/test/widgets/fade_in_image_test.dart

```diff
@@ -262,6 +265,44 @@ Future<void> main() async {
       expect(findFadeInImage(tester).target.opacity, moreOrLessEquals(1));
     });
 
+    testWidgets('memory placeholder cacheWidth and cacheHeight is passed through', (WidgetTester tester) async {
+      final Uint8List testBytes = Uint8List.fromList(kTransparentImage);
+      final FadeInImage image = FadeInImage.memoryNetwork(
+        placeholder: testBytes,
+        image: 'test.com',
+        placeholderCacheWidth: 20,
+        placeholderCacheHeight: 30,
+        imageCacheWidth: 40,
+        imageCacheHeight: 50,
+      );
+
+      final DecoderCallback decode = (Uint8List bytes, {int cacheWidth, int cacheHeight}) {
+        expect(cacheWidth, 20);
+        expect(cacheHeight, 30);
+        return PaintingBinding.instance.instantiateImageCodec(bytes, cacheWidth: cacheWidth, cacheHeight: cacheHeight);
+      };
+      final ResizeImage resizeImage = image.placeholder;
+      resizeImage.load(await resizeImage.obtainKey(ImageConfiguration.empty), decode);
+    });
+
+    testWidgets('do not resize when null cache dimensions', (WidgetTester tester) async {
+      final Uint8List testBytes = Uint8List.fromList(kTransparentImage);
+      final FadeInImage image = FadeInImage.memoryNetwork(
+        placeholder: testBytes,
+        image: 'test.com',
+      );
+
+      final DecoderCallback decode = (Uint8List bytes, {int cacheWidth, int cacheHeight}) {
+        expect(cacheWidth, null);
+        expect(cacheHeight, null);
+        return PaintingBinding.instance.instantiateImageCodec(bytes, cacheWidth: cacheWidth, cacheHeight: cacheHeight);
+      };
+      // image.placeholder should be an instance of MemoryImage instead of ResizeImage
+      final MemoryImage memoryImage = image.placeholder;
+      expect(image.placeholder is MemoryImage, true);
```

Comment:
`expect(image.placeholder, isA<MemoryImage>)`

Also, the type in the line above should probably be ImageProvider instead of MemoryImage. Otherwise you may never get to this expect.